export/export_handler.go

```diff
@@ -41,6 +49,121 @@ func PathToPointer(path string) (block.ObjectPointer, error) {
 		Identifier:       u.Path,
 	}, err
 }
+
+func (h *Handler) start(body *string) error {
+	var startData StartData
+	err := json.Unmarshal([]byte(*body), &startData)
+	if err != nil {
+		return err
+	}
+
+	config, err := h.cataloger.GetExportConfigurationForBranch(startData.Repo, startData.Branch)
+	if err != nil {
+		return err
+	}
+
+	finishBodyStr, err := getFinishBodyString(startData.Repo, startData.Branch, startData.ToCommitRef)
+	if err != nil {
+		return err
+	}
+	return h.generateTasks(startData, config, &finishBodyStr)
+}
+
+func (h *Handler) generateTasks(startData StartData, config catalog.ExportConfiguration, finishBodyStr *string) error {
+	tasksGenerator := NewTasksGenerator(startData.ExportID, config.Path, getGenerateSuccess(config.LastKeysInPrefixRegexp), finishBodyStr)
+	var diffs catalog.Differences
+	var err error
+	hasMore := true
+	after := ""
+	limit := -1
+	diffFromBase := startData.FromCommitRef == ""
+	for hasMore {
+		if diffFromBase {
+			diffs, hasMore, err = getDiffFromBase(context.Background(), startData.Repo, startData.ToCommitRef, after, limit, h.cataloger)
+		} else {
+			// Todo(guys) change this to work with diff iterator once it is available outside of cataloger
+			diffs, hasMore, err = h.cataloger.Diff(context.Background(), startData.Repo, startData.ToCommitRef, startData.FromCommitRef, catalog.DiffParams{
+				Limit:            limit,
+				After:            after,
+				AdditionalFields: []string{"physical_address"},
+			})
+		}
+		if err != nil {
+			return err
+		}
+		taskData, err := tasksGenerator.Add(diffs)
+		if err != nil {
+			return err
+		}
+		// add taskData tasks
+		err = h.parade.InsertTasks(context.Background(), taskData)
+		if err != nil {
+			return err
+		}
+		if hasMore && len(diffs) > 0 {
```

Comment:
Can we have `hasMore && len(diffs) == 0`?!  (Note that in that case it looks like an infinite loop, so not sure what this extra caution does here.  If it's a worry, it should be an error message -- maybe even fatal.)

Reply:
I missed that,
Thanks